src/H5Eint.c

```diff
@@ -658,22 +1410,13 @@ H5E_printf_stack(H5E_t *estack, const char *file, const char *func, unsigned lin
     va_start(ap, fmt);
     va_started = true;
 
-    /* Use the vasprintf() routine, since it does what we're trying to do below */
-    if (HDvasprintf(&tmp, fmt, ap) < 0)
```

Comment:
Rather than call vasprintf() here, then have H5E__push_stack() duplicate the string, just have H5E__push_stack() call vasprintf().